src/ext/serial.cpp

```diff
@@ -4,17 +4,44 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#ifdef __MBED__
+#ifndef UNIT_TEST
 
-#include "mbed.h"
 #include "config.h"
 #include "thingset.h"
 #include "ext.h"
 
+#ifdef __MBED__
+
+#include "mbed.h"
+
+#elif defined(__ZEPHYR__)
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <string.h>
+#include <stdio.h>
+#include <drivers/uart.h>
+
+#define UART_DEVICE_NAME DT_ALIAS_UART_DBG_LABEL
+struct device *uart_dev = device_get_binding(UART_DEVICE_NAME);
+
+static char buf_resp[1000];
```

Comment:
Now we have `buf_resp` defined twice. Static global here and private member of ThingSetStream class, each consuming 1 kB of memory. One should be enough.

Also all the other variables like `buf_req`, `req_pos` and `command_flag` are defined twice. I prefer keeping them as private class members.